pkg/controller/opentelemetrycollector/reconcile_serviceaccount_test.go

```diff
@@ -0,0 +1,168 @@
+	assert.Equal(t, s.Labels["custom-label"], "custom-value")
+	assert.Equal(t, s.Labels["app.kubernetes.io/name"], s.Name)
+}
+
+func TestOverrideServiceAccountName(t *testing.T) {
+	// test
+	overridden := *instance
+	overridden.Spec.ServiceAccount = "custom-sa"
+
+	ctx := context.WithValue(context.Background(), opentelemetry.ContextInstance, &overridden)
+	ctx = context.WithValue(ctx, opentelemetry.ContextLogger, logf.Log.WithName("unit-tests"))
+
+	s := ServiceAccountNameFor(ctx)
+
+	// verify
+	assert.Equal(t, "custom-sa", s)
+}
+
+func TestUnmanagedServiceAccount(t *testing.T) {
+	// customize the CR, to override the service account name
+	overridden := *instance
+	overridden.Spec.ServiceAccount = "custom-sa"
+
+	// build a context with an instance of our custom CR
+	ctx := context.WithValue(context.Background(), opentelemetry.ContextInstance, &overridden)
+	ctx = context.WithValue(ctx, opentelemetry.ContextLogger, logf.Log.WithName("unit-tests"))
+
+	c := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        overridden.Spec.ServiceAccount,
+			Namespace:   instance.Namespace,
+			Annotations: instance.Annotations,
+		},
+	}
+
+	clients := &client.Clientset{
+		Kubernetes:    fake.NewSimpleClientset(c),
+		Monitoring:    fakemon.NewSimpleClientset(),
+		OpenTelemetry: fakeotclient.NewSimpleClientset(instance),
+	}
+	reconciler := New(schem, clients)
+
+	// sanity check
+	persisted, err := clients.Kubernetes.CoreV1().ServiceAccounts(instance.Namespace).Get(c.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.NotNil(t, persisted)
+
+	// test
+	reconciler.reconcileServiceAccount(ctx)
+
+	// verify that the service account still exists and is not with the operator's labels
+	existing, err := clients.Kubernetes.CoreV1().ServiceAccounts(instance.Namespace).Get(c.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Nil(t, existing.Labels)
+
+	// verify that a service account with the name that the operator would create does *not* exist
+	managed := serviceAccount(ctx)
+	managed, err = clients.Kubernetes.CoreV1().ServiceAccounts(instance.Namespace).Get(managed.Name, metav1.GetOptions{})
+	assert.Error(t, err) // not found
+	assert.Nil(t, managed)
+}
+
+func TestProperReconcileServiceAccount(t *testing.T) {
+	// prepare
+	clients := &client.Clientset{
+		Kubernetes:    fake.NewSimpleClientset(),
+		Monitoring:    fakemon.NewSimpleClientset(),
+		OpenTelemetry: fakeotclient.NewSimpleClientset(instance),
+	}
+	reconciler := New(schem, clients)
+	req := reconcile.Request{NamespacedName: types.NamespacedName{Namespace: instance.Namespace}}
+
+	// test
+	reconciler.Reconcile(req)
+
+	// verify
+	list, err := clients.Kubernetes.CoreV1().ServiceAccounts(instance.Namespace).List(metav1.ListOptions{})
+	assert.NoError(t, err)
+
+	// we assert the correctness of the service in another test
+	assert.Len(t, list.Items, 1)
+
+	// we assert the correctness of the reference in another test
```

Comment:
Which test? (and above)

Reply:
Owner reference: as it's a generic check, it's done at `reconcile_test.go`
`list.Items`: it's checked at `TestProperServiceAccount`. I fixed the comment to say `service account` instead of `service`.